http-transport-binding.md

```diff
@@ -0,0 +1,338 @@
+
+For instance, if the declared `contentType` is
+`application/json;charset=utf-8`, the expectation is that the `data` attribute
+value is made available as [UTF-8][RFC3629] encoded JSON text to HTTP.
+
+## 3. HTTP Message Mapping
+
+The event binding is identical for both HTTP request and response messages.
+
+The content mode is chosen by the sender of the event, which is either the
+requesting or the responding party. Gestures that might allow solicitation of
+events using a particular mode might be defined by an application, but are not
+defined here.
+
+The receiver of the event can distinguish between the two modes by inspecting
+the `Content-Type` header value. If the value is prefixed with the Cloud Events
+media type `application/cloudevents`, indicating the use of a known [event
+format](#14-Event-Formats), the receiver uses *structured* mode, otherwise it
+defaults to *binary* mode.
+
+If a receiver detects the Cloud Events media type, but with an event format that
+it cannot handle, for instance `application/cloudevents+avro`, it MAY still
+treat the event as binary and forward it to another party as-is.
+
+### 3.1. Binary Content Mode
+
+The *binary* content mode accommodates any shape of event data, and allows for
+efficient transfer and without transcoding effort.
+
+#### 3.1.1. HTTP Content-Type
+
+For the *binary* mode, the HTTP `Content-Type` value maps directly to the Cloud
+Events `contentType` attribute.
+
+#### 3.1.2. Event Data Encoding
+
+The [`data` attribute](#22-data-attribute) byte-sequence is used as the HTTP
+message body.
+
+#### 3.1.3. Metadata Headers
+
+All [Cloud Events][CE] attributes with exception of `contentType` and `data`
+are individually mapped to and from distinct HTTP message headers.
+
+##### 3.1.3.1 HTTP Header Names
+
+The naming convention for the HTTP header mapping of attributes is:
+
+    * Prefixed with "CE-"
+    * Each attribute name's first character is capitalized
+
+Examples:
+
+    * `eventTime` maps to `CE-EventTime`
+    * `eventId` maps to `CE-EventId`
+    * `cloudEventsVersion` maps to `CE-CloudEventsVersion`
+
+##### 3.1.3.2 HTTP Header Values
+
+The value for each HTTP header is constructed from the respective attribute's
+[JSON value][JSON-value] representation, compliant with the [JSON event
+format][JSON-format] specification.
+
+The JSON value MUST NOT contain any unencoded LF (ASCII 0x10) or CR (ASCII
+0x13) characters.
+
+JSON objects and arrays are NOT surrounded with single or double quotes.
+
+#### 3.1.4 Examples
+
+This example shows the *binary* mode mapping of an event with an HTTP POST
+request:
+
+``` text
+POST /someresource HTTP/1.1
+Host: webhook.example.com
+CE-CloudEventsVersion: "0.1"
+CE-EventType: "com.example.someevent"
+CE-EventTime: "2018-04-05T03:56:24Z"
+CE-EventId: "1234-1234-1234"
+CE-Source: "/mycontext/subcontext"
+    .... further attributes ...
+Content-Type: application/json; charset=utf-8
+Content-Length: nnnn
+
+{
```

Comment:
I wonder if we should remove the {}'s otherwise people might thing that {} isn't part of the app data, and I believe in this example it is.

Reply:
That's why I broke those across multiple lines. I find the notation here somewhat better than adding fake data that people will then argue about.